security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -220,9 +220,11 @@ func (sc *SecretCache) keyCertRotationJob() {
 	}
 }
 
-func (sc *SecretCache) rotate(t time.Time) {
+func (sc *SecretCache) rotate() {
 	log.Debug("Refresh job running")
 
+	secretMap := map[ConnKey]*model.SecretItem{}
+	wg := sync.WaitGroup{}
```

Comment:
Original logic indeed has a potential issue: old rotate goroutine may not exit when new rotate takes place